icefoganalytics/sfa-client
# Enhancement: Add Student Dependents Section to Student Application Endpoint Response

Relates to https://github.com/icefoganalytics/student-financial-aid/issues/2
Fixes https://github.com/icefoganalytics/sfa-client/issues/28

# Context

As part of the "view submitted applications and their current status" project, this task covers returning the student dependents data.

# Implementation

Load dependents and dependent eligibilities.
Refactor the student application service into its own file. Its now a more specific retrieval service than the generic application service. This makes room for a future where we might server a less customized application blob.
Add a `dev console` to quickly access a typescript node console for manual testing.

# Concerns

The front-end assumes that the dependent <-> dependent eligibility relationship is one-to-one.
However, there isn't a constraint forcing that constraint at the back-end. In fact, each dependent can have multiple eligibilities, _and do_ in the database. I don't think it makes sense to have a one-to-many relationship in the database, so in the future, we will probably want to enforce the one-to-one constraint, and clean up the data.

# Data Format

```json
{
  "id": "some-application-id",
  "studentDependants": { "dependants": [] },
}
```

# Testing Instructions

1. Boot the `sfa-client` repo via `API_PORT=3100 dev up`
2. Boot the `student-financial-aid` front- and back-ends.
3. Log in to the app at http://localhost:8080/
4. Go to http://localhost:3000/api/portal/students/1/applications/2198 and you will see the mostly blank student dependants section. Note: the front-end spells it "dependants" and the back-end (and database) spells it "dependents". I opted to copy the database spelling as that is harder to change.
5. Go to http://localhost:3000/api/portal/students/582/applications/319 and you will see that studentDependants section some content.
